packages/web-components/src/dialog/dialog.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { expect, test } from '@playwright/test';
+import { fixtureURL } from '../helpers.tests.js';
+
+test.describe('Dialog', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(fixtureURL('components-dialog-dialog--default'));
+
+    await page.waitForFunction(() => customElements.whenDefined('fluent-button'));
+  });
+
+  test('should open and close dialog', async ({ page }) => {
+    const openButton = await page.$('fluent-button');
+    await openButton?.click();
+
+    const dialog = await page.$('fluent-dialog');
+    expect(dialog).not.toBeNull();
+
+    const closeButton = await dialog?.$('fluent-button');
+    await closeButton?.click();
+
+    await expect(page.locator('fluent-dialog')).toBeHidden();
+  });
```

Comment:
Typically for tests I prefer if the story content be ignored and overridden with `page.setContent()`.

Here's how I'd approach this test:
```ts
test('should show and hide the dialog content', async ({ page }) => {
  // Locator objects are references which are re-evaluated each time they're used
  const element = page.locator('fluent-dialog');
  const content = element.locator('#content');

  // Using `setContent` ensures that the only things on the page are what the test needs.
  // It also avoids issues if the default story fixture itself ever changes.
  await page.setContent(/* html */ `
    <fluent-dialog>
        <div id="content">content</div>
    </fluent-dialog>
  `);

  // this particular test can be split into multiple steps to improve clarity
  await test.step('should show the dialog content when the dialog is shown', async () => {
    await expect(content).toBeHidden();

    await element.evaluate((node: Dialog) => {
      node.show();
    });

    await expect(content).toBeVisible();
  });

  await test.step('should hide the dialog content when the dialog is hidden', async () => {
    await element.evaluate((node: Dialog) => {
      node.hide();
    });

    await expect(content).toBeHidden();
  });
});
```

It would be great to see more tests here, primarily around clicking to hide the dialog after it's shown.

Reply:
@radium-v This was super helpful, thanks! I've beefed up the tests for `dialog` and `dialog-body`. Take a look and let me know if it looks good enough. 🚀